src/main/java/aRC/TaskList.java

```diff
@@ -0,0 +1,78 @@
+package arc;
+
+import java.util.ArrayList;
+
+/**
+ * Encapsulates a list of Tasks
+ */
+public class TaskList {
+
+    private ArrayList<Task> arrayList;
+
+    /**
+     * Constructor for TaskList
+     * @param arrayList An ArrayList of Tasks
+     */
+    public TaskList(ArrayList<Task> arrayList) {
+        this.arrayList = arrayList;
+    }
+
+    /**
+     * Returns the number of tasks currently in TaskList
+     * @return The number of tasks currently in TaskList
+     */
+    public int numTasks() {
+        return this.arrayList.size();
+    }
+
+    /**
+     * Gets the tasks at the given index
+     * @param index The index of the task to retrieve
+     * @return The task at the given index
+     */
+    public Task getTask(int index) {
+        return arrayList.get(index);
+    }
+
+    /**
+     * Prints out all Tasks whose title matches keyword
+     * @param keyword Keyword of Task
+     */
+    public void listTasks(String keyword) {
+        System.out.println(String.format(
+                "Listing the%s tasks in your list...", !keyword.equals("") ? " matching" : ""
+        ));
+
+        if (this.arrayList.size() == 0) {
+            System.out.println("You have no current tasks :-(");
+        }
+
+        for (int i = 0; i < this.arrayList.size(); i++) {
+            if (this.arrayList.get(i).title.toLowerCase().contains(keyword.toLowerCase())) {
```

Comment:
Perhaps you can declare this predicate as a boolean variable with a name as it is not immediately obvious what it does